consul/aio.py

```diff
@@ -21,35 +21,47 @@ def __init__(self, *args, loop=None, connections_limit=None, connections_timeout
         if connections_timeout:
             timeout = aiohttp.ClientTimeout(total=connections_timeout)
             session_kwargs["timeout"] = timeout
+
+        header = self.build_header(**kwargs)
+        if header:
+            session_kwargs["headers"] = header
         self._session = aiohttp.ClientSession(connector=connector, **session_kwargs)
 
-    async def _request(self, callback, method, uri, data=None, connections_timeout=None):
+    async def _request(self, callback, method, uri, data=None, connections_timeout=None, **kwargs):
```

Comment:
shouldn't we explicitly pass and propagate a "headers" argument instead?
kwargs is quite lax, consequently you won't be able to do that for another use-case in the future.

Reply:
agree

Reply:
@mbrulatout any blocker on this?

Reply:
Nope, it just got lost.
Not sure about the scope of the comment though. Is it line/file specific or global ?
Assuming it's about getting rid of **kwargs everywhere it's been introduced, I'm not a big fan.
I suppose we could keep the **kwargs up to the get/put/post/delete, which would be in charge of building and passing the "headers" arg to _request(...)